tests/checkpointing/test_model_checkpoint.py

```diff
@@ -1020,3 +1020,46 @@ def __init__(self, hparams):
     else:
         # make sure it's not AttributeDict
         assert type(ckpt[model.CHECKPOINT_HYPER_PARAMS_KEY]) == hparams_type
+
+
+@pytest.mark.parametrize('max_epochs', [3, 4])
+@pytest.mark.parametrize(
+    'save_top_k, expected',
+    [
+        (1, ['curr_epoch.ckpt']),
+        (2, ['curr_epoch.ckpt', 'curr_epoch-v0.ckpt']),
+    ]
+)
+def test_model_checkpoint_file_already_exists(tmpdir, max_epochs, save_top_k, expected):
+    """
+    Test that version is added to filename if required and it already exists in dirpath.
+    """
+    class CustomModel(LogInTwoMethods):
+        def validation_epoch_end(self, outputs):
+            return {'epoch': self.current_epoch}
```

Comment:
You don't need to subclass `BoringModel` to monitor `epoch`. You can do it already with `BoringModel()`. See these tests for example:

https://github.com/PyTorchLightning/pytorch-lightning/pull/5008/files#diff-44594fac97ae772bb165301c22fdc1b25cd481589502dd7413c6953289cbb718R1033-R1044

Reply:
yeah, I was actually monitoring `val_acc` initially.